src/modules/welcome/Config.cpp

```diff
@@ -261,13 +268,47 @@ QString Config::supportUrl() const
 void Config::setSupportUrl(const QString& url)
 {
     m_supportUrl = url;
+    emit supportUrlChanged();
 }
 
+void RequirementsModel::retranslate()
+{
+    if ( !m_satisfiedRequirements )
+    {
+        QString message;
+        const bool setup = Calamares::Settings::instance()->isSetupMode();
 
+        if ( !m_satisfiedMandatory )
+        {
+            message = setup ? tr( "This computer does not satisfy the minimum "
+            "requirements for setting up %1.<br/>"
+            "Setup cannot continue. "
+            "<a href=\"#details\">Details...</a>" )
+            : tr( "This computer does not satisfy the minimum "
+            "requirements for installing %1.<br/>"
+            "Installation cannot continue. "
+            "<a href=\"#details\">Details...</a>" );
+
+        }else
+        {
+            message = setup ? tr( "This computer does not satisfy some of the "
+            "recommended requirements for setting up %1.<br/>"
+            "Setup can continue, but some features "
+            "might be disabled." )
+            : tr( "This computer does not satisfy some of the "
+            "recommended requirements for installing %1.<br/>"
+            "Installation can continue, but some features "
+            "might be disabled." );
+        }
 
+        m_warningMessage = message.arg( *Calamares::Branding::ShortVersionedName );
+    }else
+    {
+        m_warningMessage = tr( "This program will ask you some questions and "
+        "set up %2 on your computer." )
+        .arg( *Calamares::Branding::ProductName );
+    }
 
-
-
-
-
+    emit warningMessageChanged();
```

Comment:
Why is the warning message part of the model, at all? This is a UI thing that should be looking at "are all the requirements met". I suppose a three-state satisfied/recommended/required enum could be part of the model for convenience.

Reply:
the model is the one being used on the qwidgets, so I thought it would be cleaner to make the model have the messages, instead of passing the Conf object down to all the qwidgets handling the modeling, that would a solution or refactor the UI and make those messages more top-level to the WelcomePage